textual_fspicker/base_dialog.py

```diff
@@ -100,11 +112,14 @@ def compose(self) -> ComposeResult:
         """
         with Dialog() as dialog:
             dialog.border_title = self._title
+            currentDrive = str(os.path.splitdrive(MakePath.of(self._location).expanduser().absolute())[0]+'\\').upper()
```

Comment:
Could you break down for me what this line of code is actually doing? For example, it looks like there's a hard-coded path separator being used here; what is the reason for that?

Reply:
shady is to take the root of the given path
and serialize it to match the select values that match self.disks

```python
disks = ["C:\\","D:\\","E:\\", ...]

# Get path absolute to _location exsample: ~\textual_fspicker
absolute_path = MakePath.of(self._location).expanduser().absolute()
# out: Path d:\project\textual_fspicker

# Split the root from the path
tuple_path = os.path.splitdrive(absolute_path)
# out: Tuple ["d:\", "project\textual_fspicker"]

# Get only root and serialize backslash
serialize_root_path = tuple_path[0]+'\\' 
# out: String "d:\\"

# Type: String as root disck
current_drive =  serialize_root_path.upper()
# out String "D:\\"

```